src/main/scala/com/mozilla/telemetry/streaming/EventsToAmplitude.scala

```diff
@@ -244,21 +276,36 @@ object EventsToAmplitude extends StreamingJobBase {
       ))
 
       val pingsDataFrame = spark.createDataFrame(pings, schema)
-      val apiKey = sys.env(AMPLITUDE_API_KEY_KEY)
-      val minDelay = opts.minDelay()
-      val url = opts.url()
-
-      log.info(s"Processing events for ${pingsDataFrame.count()} pings on $currentDate")
-
-      getEvents(config, pingsDataFrame, opts.sample(), opts.raiseOnError())
-        .repartition(maxParallelRequests)
-        .foreachPartition{ it: Iterator[String] =>
-          val httpSink = AmplitudeHttpSink(apiKey = apiKey, url = url)
-          it.foreach{ event =>
-            httpSink.process(event)
-            java.lang.Thread.sleep(minDelay)
+
+      infoAndPrintln(s"Processing events for ${pingsDataFrame.count()} pings on $currentDate")
+
+      import spark.implicits._
+
+      val keyedDataset: Dataset[KeyedAmplitudePayload] =
+        getEvents(config, pingsDataFrame, opts.sample(), opts.raiseOnError())
+          .repartition(maxParallelRequests, f.col("clientId"))  // Bug 1484819
```

Comment:
Note that we're partitioning on the clientId field of KeyedAmplitudePayload now.